spec/support/capybara/block_requests_made_outside_of_test.rb

```diff
@@ -0,0 +1,60 @@
+# When test is finished, Capybara calls `Capybara.reset!` which in turn calls
+# `driver.reset!`. This one is responsible for stopping the browser by
+# navigating to about:blank page and waiting for pending requests to complete.
+#
+# With Cuprite, we observed that some requests could still be triggered from the
+# browser after `Capybara.reset!` is called. It can interfere with test
+# execution is some unexpected ways: when a request to the API is made, the
+# settings are read from the database. If this happens right when the database
+# is being rolled back to a previous savepoint (which happens when using
+# `before_all` helper), then in postgres dapater code there is a `nil` reference
+# instead of a result, and then it errs with "NoMethodError: undefined method
+# 'clear' for nil".
+#
+# You can run tests from `spec/features/work_packages/progress_modal_spec.rb` a
+# couple of times to experiment the error. It happens mostly with tests having
+# the most nested `before_all` calls.
+#
+# We tried navigating to about:blank with Cuprite, but some requests were still
+# made. So we looked for another fix.
+#
+# Using a middleware to actively block requests outside of test execution fixed
+# the issue.
+
+class RequestsBlocker
+  class << self
+    attr_accessor :blocked
+  end
```

Comment:
Because I initially called `RequestsBlocker.block_requests!` and changed to `Capybara.app.block_requests!` but did not change the attribute.

Good point, I'll move it to instance level.